client/scraper/general_metadata.py

```diff
@@ -1,21 +1,59 @@
-from HTMLParser import HTMLParser
-import json
-
 from metadata import Metadata
+from client.constants import JsonFields, MetadataFields
+from bs4 import BeautifulSoup
+
 
+class GeneralMetadata(Metadata):
+    prop_map = Metadata.prop_map
 
-class GeneralMetadata(Metadata, HTMLParser):
-    json_return = ""
-    prop_map = {}
+    def __init__(self):
+        Metadata.__init__(self)
 
     def parse_content(self, content):
-        self.feed(content.content)
+        soup = BeautifulSoup(content.content)
+        title = self.fetch_title(soup)
+        desc = self.fetch_desc(soup)
+        image_url = self.fetch_image_url(soup)
+
+        if title:
+            self.prop_map[JsonFields.TITLE] = title
+
+        if desc:
+            self.prop_map[JsonFields.DESC] = desc
+
+        if image_url:
+            media_list = {}
+            media_list[JsonFields.COUNT] = 1
+            media_list[JsonFields.DATA] = []
+            data = {
+                JsonFields.TYPE: JsonFields.IMAGE,
+                JsonFields.SRC: image_url
+            }
+            media_list[JsonFields.DATA].append(data)
+            self.prop_map[JsonFields.MEDIA] = media_list
         return self.to_json(self.prop_map)
 
-    def handle_starttag(self, tag, attrs):
-        if tag == 'meta' and len(attrs) > 1:
-            key = attrs[0][1]
-            self.prop_map[key] = attrs[1][1]
+    def fetch_title(self, soup):
```

Comment:
I think we should name this
get_title
not fetch... since fetching is associated with downloading from online something and we've already fetched the data
(same with all other fetch_*)